### PR TITLE
Email confirmation checkbox

### PR DESCRIPTION
**Pull request by andrzejewsky**
closes: https://github.com/saleor/saleor-dashboard/issues/3653

### Screenshots
Screenshot 2023-05-22 at 10 19 30

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [x] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

File: src/siteSettings/components/SiteSettingsPage/SiteSettingsPage.tsx [hunk 1]
```diff
@@ -15,8 +15,8 @@ import useStateFromProps from "@dashboard/hooks/useStateFromProps";
 import { commonMessages } from "@dashboard/intl";
 import createSingleAutocompleteSelectHandler from "@dashboard/utils/handlers/singleAutocompleteSelectChangeHandler";
 import { mapCountriesToChoices } from "@dashboard/utils/maps";
-import { ConfirmButtonTransitionState, makeStyles } from "@saleor/macaw-ui";
-import { Box } from "@saleor/macaw-ui/next";
+import { ConfirmButtonTransitionState } from "@saleor/macaw-ui";
```
Comment (krzysztofzuraw) [inline src/siteSettings/components/SiteSettingsPage/SiteSettingsPage.tsx:18]: **suggestion**: I saw in @poulch PR that we migrated `ConfirmButton` to new MacawUI in dashboard. 

@poulch has this PR been merged?

File: src/warehouses/components/WarehouseInfo/WarehouseInfo.tsx [hunk 2]
```diff
@@ -1,10 +1,10 @@
-import CardTitle from "@dashboard/components/CardTitle";
+import { DashboardCard } from "@dashboard/components/Card";
 import { WarehouseErrorFragment } from "@dashboard/graphql";
 import { FormChange } from "@dashboard/hooks/useForm";
 import { commonMessages } from "@dashboard/intl";
 import { getFormErrors } from "@dashboard/utils/errors";
 import getWarehouseErrorMessage from "@dashboard/utils/errors/warehouse";
-import { Card, CardContent, TextField } from "@material-ui/core";
+import { TextField } from "@material-ui/core";
```
Comment (krzysztofzuraw) [inline src/warehouses/components/WarehouseInfo/WarehouseInfo.tsx:7]: **suggestion**: you can use new `Input` from MacawUI here

File: src/siteSettings/components/SiteSettingsPage/messages.ts [hunk 3]
```diff
@@ -23,4 +23,26 @@ export const messages = defineMessages({
...
+  },
+  sectionEmailConfirmationOption: {
+    id: "uRfta0",
+    defaultMessage: "Enable email confirmation",
+    description: "Enable email confirmation",
+  },
+  sectionEmailConfirmationDescription: {
+    id: "GpyFuP",
+    defaultMessage:
+      "Require an confirmation for newly registered users by using their email.",
```
Comment (Droniu) [inline src/siteSettings/components/SiteSettingsPage/messages.ts:44]: **typo**
```suggestion
      "Require a confirmation for newly registered users by using their email.",
```

**Review by Droniu — APPROVED** [review]

**Comment by timuric** [comment]
Missing bottom padding 
image

**Review by Droniu — DISMISSED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Review by wojteknowacki — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Review by krzysztofzuraw — APPROVED** [review]

**Pull request merged**